awx_collection/plugins/modules/tower_notification.py

```diff
@@ -26,168 +26,188 @@
     name:
       description:
         - The name of the notification.
+      type: str
       required: True
+    new_name:
+      description:
+        - Setting this option will change the existing name (looked up via the name field.
       type: str
     description:
       description:
         - The description of the notification.
-      required: False
       type: str
     organization:
       description:
         - The organization the notification belongs to.
-      required: False
       type: str
     notification_type:
       description:
         - The type of notification to be sent.
       required: True
-      choices: ["email", "slack", "twilio", "pagerduty", "hipchat", "webhook", "irc"]
+      choices:
+        - 'email'
+        - 'grafana'
+        - 'hipchat'
+        - 'irc'
+        - 'mattermost'
+        - 'pagerduty'
+        - 'rocketchat'
+        - 'slack'
+        - 'twilio'
+        - 'webhook'
       type: str
     notification_configuration:
       description:
         - The notification configuration file. Note providing this field would disable all notification-configuration-related fields.
-      required: False
-      type: str
+      type: dict
+    messages:
+      description:
+        - Optional custom messages for notification template.
+      type: dict
+      default:
+        started:
+        success:
+        error:
+        workflow_approval:
```

Comment:
This is a case (along with the notification_configuration) I think that the `sample` feature might be really nice. Example:

https://github.com/ansible/ansible/blob/e3591223a042caa537fcc88cf7553227b8657f70/test/support/integration/plugins/modules/ec2_eni_info.py#L57

In the real world, almost everyone starts by copying and pasting a sample, and this lets us put it in the doc spec, so users are almost sure to see it.

Reply:
Added an example of adding custom messages to a Slack notification type in the ansible documentation.

Reply:
Let me know if you think we should add them for all of the different notification types.

Reply:
Can we remove the default? I just don't see any good coming out of it.

Reply:
Yes, done. I didn't realize that these were nested dicts.